AndroidOnChromeOSDragAndDropDemo/app/src/main/java/dev/hadrosaur/draganddropsample/MainActivity.kt

```diff
@@ -0,0 +1,172 @@
+            return when(event.action) {
+                ACTION_DRAG_STARTED -> {
+                    if (event.clipDescription.hasMimeType(MIMETYPE_TEXT_PLAIN)
+                        || event.clipDescription.hasMimeType("image/png")) {
+
+                        // Greenify background colour so user knows this is a target
+                        view.setBackgroundColor(getColor(android.R.color.holo_green_dark))
+                        return true
+                    }
+                    false
+                }
+                ACTION_DRAG_ENTERED -> {
+                    // More intense green background colour when item is over top of target
+                    view.setBackgroundColor(getColor(android.R.color.holo_green_light))
+                    true
+                }
+                ACTION_DRAG_LOCATION -> { true }
+                ACTION_DRAG_EXITED -> {
+                    // Less intense green background colour when item not over target
+                    view.setBackgroundColor(getColor(android.R.color.holo_green_dark))
+                    true
+                }
+                ACTION_DRAG_ENDED -> {
+                    // Back to default colour when drag is not in process
+                    view.setBackgroundColor(getColor(android.R.color.holo_purple))
+                    true
+                }
+                ACTION_DROP -> {
+                    requestDragAndDropPermissions(event) //Allow items from other applications
+                    val item = event.clipData.getItemAt(0)
+
+                    if (event.clipDescription.hasMimeType(MIMETYPE_TEXT_PLAIN)) {
+                        if (item.text != null) {
+                            // Plain text item, show it on screen
+                            binding.textDropTarget.setTextSize(COMPLEX_UNIT_SP, 22f)
+                            binding.textDropTarget.text = item.text
+                        } else {
+                            // This is a plain text file. Read some characters and show them
+                            // Use ContentResolver to resolve the URI
+                            val contentUri = item.uri
+                            var parcelFileDescriptor: ParcelFileDescriptor? = null
+                            try {
+                                parcelFileDescriptor =
+                                    contentResolver.openFileDescriptor(contentUri, "r")
+                            } catch (e: FileNotFoundException) {
+                                e.printStackTrace()
+                                logd("File not found.")
+                                return false
+                            }
+
+                            if (parcelFileDescriptor == null) {
+                                logd("Error: could not load file: ${contentUri.toString()}")
+                                binding.textDropTarget.text = "Error: could not load file: ${contentUri.toString()}"
```

Comment:
In this case, `textDropTarget.setTextSize()` is not set as in the others, which can result in large or small text sizes, depending on the previous drag event.